src/nft/components/collection/TransactionCompleteModal.tsx

```diff
@@ -0,0 +1,363 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { formatEther } from '@ethersproject/units'
+import clsx from 'clsx'
+import { Box } from 'nft/components/Box'
+import { Portal } from 'nft/components/common/Portal'
+import { Row } from 'nft/components/Flex'
+import { BackArrowIcon, ChevronUpIcon, LightningBoltIcon, UniIcon } from 'nft/components/icons'
+import { Overlay, stopPropagation } from 'nft/components/modals/Overlay'
+import { vars } from 'nft/css/sprinkles.css'
+import { useIsMobile, useSendTransaction, useTransactionResponse } from 'nft/hooks'
+import { TxResponse, TxStateType, UpdatedGenieAsset } from 'nft/types'
+import { fetchPrice, formatEthPrice, formatUsdPrice, formatUSDPriceWithCommas, shortenTxHash } from 'nft/utils'
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { ExplorerDataType, getExplorerLink } from 'utils/getExplorerLink'
+
+import * as styles from './TransactionCompleteModal.css'
+
+const getTotalNftValue = (nfts: UpdatedGenieAsset[]): BigNumber => {
+  return (
+    nfts &&
+    nfts.reduce(
+      (ethTotal, nft) =>
+        ethTotal.add(BigNumber.from(nft.updatedPriceInfo ? nft.updatedPriceInfo.ETHPrice : nft.priceInfo.ETHPrice)),
+      BigNumber.from(0)
+    )
+  )
+}
+
+const TxCompleteModal = () => {
+  const [ethPrice, setEthPrice] = useState(3000)
+  const [showUnavailable, setShowUnavailable] = useState(false)
+  const txHash = useSendTransaction((state) => state.txHash)
+  const setTxState = useSendTransaction((state) => state.setState)
+  const txState = useSendTransaction((state) => state.state)
+  const transactionStateRef = useRef(txState)
+  const transactionResponse = useTransactionResponse((state) => state.transactionResponse)
+  const setTransactionResponse = useTransactionResponse((state) => state.setTransactionResponse)
+  const isMobile = useIsMobile()
+  const txHashUrl = getExplorerLink(1, txHash, ExplorerDataType.TRANSACTION)
+  const shouldShowModal = (txState === TxStateType.Success || txState === TxStateType.Failed) && txState
+  const {
+    nftsPurchased,
+    nftsNotPurchased,
+    showPurchasedModal,
+    showRefundModal,
+    totalPurchaseValue,
+    totalRefundValue,
+    totalUSDRefund,
+    txFeeFiat,
+  } = useMemo(() => {
```

Comment:
this file would be a lot cleaner if we could move this useMemo to a helper function, something like 'parseTransactionResponse'